cobra/core/Model.py

```diff
@@ -266,9 +421,49 @@ def optimize(self, objective_sense='maximize', **kwargs):
                    specified with the appropriate keyword argument.
 
         """
-        solution = optimize(self, objective_sense=objective_sense, **kwargs)
-        self.solution = solution
-        return solution
+        if kwargs.get('solver', None) in ('cglpk', 'gurobi'):
+            solution = optimize(self, objective_sense=objective_sense,
+                                **kwargs)
+            self.solution = solution
+            return solution
+
+        else:  # make the following honor the solver kwarg ...
+            # from cameo ...
+            self._timestamp_last_optimization = time.time()
+            if objective_sense is not None:
+                original_direction = self.solver.objective.direction
+                self.solver.objective.direction = \
+                    {'minimize': 'min', 'maximize': 'max'}[objective_sense]
+                self.solver.optimize()
+                self.solver.objective.direction = original_direction
+            else:
+                self.solver.optimize()
+            solution = solution_type(self)
+            self.solution = solution
+            return solution
+
+    def solve(self, solution_type=LazySolution, *args, **kwargs):
```

Comment:
This seems confusing to me. Why do we need a `model.optimize` and a `model.solve`?